iroh-net/src/magicsock/endpoint.rs

```diff
@@ -249,45 +251,17 @@ impl Endpoint {
 
         // If we found a candidate, set to best addr
         if let Some(pong) = last_pong {
-            if self.best_addr.is_none() {
-                // we now have a direct connection, adjust direct connection count
-                inc!(MagicsockMetrics, num_direct_conns_added);
-                if self.derp_region.is_some() {
-                    // we no longer rely on the relay connection, decrease the relay connection
-                    // count
-                    inc!(MagicsockMetrics, num_relay_conns_removed);
-                }
-            }
-
-            let addr = pong.from.as_socket_addr();
-            let trust_best_addr_until = pong.pong_at + Duration::from_secs(60 * 60);
-
-            info!(
-               %addr,
-               latency = ?lowest_latency,
-               trust_for = ?trust_best_addr_until.duration_since(Instant::now()),
-               "new best_addr (candidate address with most recent pong)"
+            self.best_addr.insert(
+                pong.from.as_socket_addr(),
```

Comment:
Ah, true that `SendAddr::as_socket_addr` maps Derp addresses to fake addresses. However, in this place I think it is correct (even though not enforced on the type level), because we only use pongs and thus `SendAddr`s out of `Endpoint::direct_addr_state`, which should only ever contain pongs for direct (UDP) addresses. So I do think that this is correct here, unless I'm missing something.

I already thought of abstracing the direct addr state in a similar manner to this PR.

In any case - if the behavior is the same as it is on `main`, let's leave this to a followup and verify there that no fake Derp addresses can ever end up in the `best_addr`.

Reply:
ah yes, I fixed that in fact. Probably the reason why it's still there is because we store a pong for the derper (in the derp_region) as well